### PR TITLE
releng: add presubmit prow job to test k8s-cloud-builder image build

### PR DESCRIPTION
**Pull request by cpanato**
k/release associate pr: https://github.com/kubernetes/release/pull/1951

/assign @justaugustus @saschagrunert @hasheddan 

tracking issue: https://github.com/kubernetes/release/issues/1850

### CODE REVIEW

File: config/jobs/kubernetes/release/release-config.yaml [hunk 1]
```diff
@@ -304,6 +304,42 @@ presubmits:
       testgrid-num-failures-to-alert: '10'
       testgrid-alert-email: release-managers+alerts@kubernetes.io
       testgrid-num-columns-recent: '30'
+  - name: pull-release-image-k8s-ci-builder
```
Comment (hasheddan) [inline config/jobs/kubernetes/release/release-config.yaml:307]: It looks like linked PR is for `k8s-cloud-builder`, but this is for `k8s-ci-builder`

Comment (cpanato) [reply config/jobs/kubernetes/release/release-config.yaml:307]: true! oh boy names confusion! good catch, will update

Comment (cpanato) [reply config/jobs/kubernetes/release/release-config.yaml]: fixed

**Pull request merged**